mainauto.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+#include "display.h"
+#include "mapio.h"
+#include "datastrc.h"
+#include "control.h"
+
+int main(int argc, char &argv){
+
+allocate_memory(*gmdt);
+//First allocate some memory
+
+allocate_pngns(*gmdt);
+//allocate memory for penguins
+
+read_data(argv[1], *gmdt);
+
+if(gmdt->crnt_player == 0){
+    generate_map(*gmdt);
+    crnt_player = 2;
+    write_data(*gmdt);
+    return 1;
+}
```

Comment:
This is not necessary, this is the auto version, so it just does one step. No need to generate map if it's already read from a file.

Just remove this entire block attached to `if(gmdt->crnt_player == 0)`